src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java

```diff
@@ -105,1002 +104,1603 @@
+                            if (medicalInventoryRow.getId() == 0) {
+                                medicalInventoryRowManager.newMedicalInventoryRow(medicalInventoryRow);
+                            } else {
+                                medicalInventoryRowManager.updateMedicalInventoryRow(medicalInventoryRow);
+                            }
+                        }
+                    }
+                    MessageDialog.info(null, "angal.inventory.update.success.msg");
+                    resetVariable();
+                    fireInventoryUpdated();
+                    int info = MessageDialog.yesNo(null, "angal.inventory.doyouwanttocontinueediting.msg");
+                    if (info != JOptionPane.YES_OPTION) {
+                        dispose();
+                    }
+                }
+            } catch (OHServiceException e) {
+                OHServiceExceptionUtil.showMessages(e);
+            }
+        });
+        return saveButton;
+    }
+
+    private JButton getValidateButton() {
+
+        validateButton = new JButton(MessageBundle.getMessage("angal.inventory.validate.btn"));
+        validateButton.setMnemonic(MessageBundle.getMnemonic("angal.inventory.validate.btn.key"));
+        validateButton.setEnabled(inventory != null);
+        validateButton.addActionListener(actionEvent -> {
+            if (inventory == null) {
+                MessageDialog.error(null, "angal.inventory.inventorymustsavebeforevalidation.msg");
+                return;
+            }
+	        int reset = MessageDialog.yesNo(null, "angal.inventory.doyoureallywanttovalidatethisinventory.msg");
+			if (reset == JOptionPane.YES_OPTION) {
+				String dischargeCode = inventory.getDischargeType();
+				String chargeCode = inventory.getChargeType();
+				Integer supplierId = inventory.getSupplier();
+				String destinationCode = inventory.getDestination();
+				if (chargeCode == null || chargeCode.isEmpty()) {
+					MessageDialog.error(null, "angal.inventory.choosechargetypebeforevalidation.msg");
+					return;
 				}
-			} else {
-				return;
+                if (dischargeCode == null || dischargeCode.isEmpty()) {
+                    MessageDialog.error(null, "angal.inventory.choosedischargetypebeforevalidation.msg");
+                    return;
+                }
+                if (supplierId == null || supplierId == 0) {
+                    MessageDialog.error(null, "angal.inventory.choosesupplierbeforevalidation.msg");
+                    return;
+                }
+                if (destinationCode == null || destinationCode.isEmpty()) {
+                    MessageDialog.error(null, "angal.inventory.choosedestinationbeforevalidation.msg");
+                    return;
+                }
+                // validate inventory
+                int inventoryRowsSize = inventoryRowSearchList.size();
+                try {
+                    medicalInventoryManager.validateMedicalInventoryRow(inventory, inventoryRowSearchList);
```

Comment:
The ward inventory cannot use the same validation method for main inventory... it makes no sense...